`SA1119` false positives on `System.Range` syntax

`SA1119` is giving false positives for `System.Range` syntax. If you make an extension method on `System.Range` (perhaps something like `ToArray()`), you must wrap the range syntax in parentheses because method calls are higher priority than range syntax. Example:
```cs
using System;

int[] array = (1..10).ToArray(); // <- false positive SA1119

static class C
{
    public static int[] ToArray(this Range range) => default;
}
```
The parentheses are required in the above example, but `SA1119` thinks they are not necessary, and when you perform the auto-correct feature of `SA1119` it removes the parentheses breaking your code with a compiler error.

> I ran into this issue with my extension methods here:
https://github.com/ZacharyPatten/Towel/blob/3e2509e887ddfacb095c9c38efdbd1deea2ec4cd/Sources/Towel/Extensions.cs#L668

@FreeApophis could you provide more context for your code snippet?
I'm trying to reproduce, but get `error CS0029: Cannot implicitly convert type 'System.Range' to 'System.Collections.Generic.IEnumerable<int>'`

```csharp
using System;
using System.Collections.Generic;

public class C {
    public void M() {
        var z = from x in (12..)
            from y in 1..3
            select x + y;
    }
}

public static class RangeExtensions
{    
    public static IEnumerable<int> SelectMany(this Range range, Func<Range, IEnumerable<int>> func)
    {
        return new int[] {0};
    }
}
```

https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEA3GUCWAZgJ5oAmIA1AD4ACATAIwCwAULQAwAEtjKA3Gza0AzD3pcAwlwDebLgp5jaKLgFkAFAEpZ8xfuwBDKFwBeXALxdCUCAFsuCLvgB2XDY3oA6L1r36A61sHYmc3Rh8Rf0DFAGcYABsYMAxHLkouYkFWfQBfNnzWYWVGJHEuACVDFwBzGABRBAwYF1j8CFa2GX1o0R5S/pEAHlcMAD4uAGVE5Iw1auINDAALfFjK6rquKE2YNB4AViGq2r3BkZdxicIAVxcwPxzFOSfA2gB2LhcYAHcwjAA2gBdWQcXLZPIFIA==

--- a/StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/MaintainabilityRules/SA1119CSharp8UnitTests.cs
+++ b/StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/MaintainabilityRules/SA1119CSharp8UnitTests.cs
@@ -170,5 +170,78 @@ public class Foo
 
             await VerifyCSharpDiagnosticAsync(LanguageVersion.CSharp8, testCode, DiagnosticResult.EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
         }
+
+        [Fact]
+        [WorkItem(3370, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3370")]
+        public async Task TestRangeFollowedByMemberCallAsync()
+        {
+            const string testCode = @"using System;
+
+public class TestClass
+{
+    public void TestMethod()
+    {
+        var array = (1..10).ToString();
+    }
+}
+";
+
+            await new CSharpTest(LanguageVersion.CSharp8)
+            {
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp31,
+                TestCode = testCode,
+            }.RunAsync(CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        [WorkItem(3370, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3370")]
+        public async Task TestRangeAsync()
+        {
+            const string testCode = @"using System;
+
+public class TestClass
+{
+    public void TestMethod()
+    {
+        var a = new int[0];
+        var b = a[(..)];
+        var range = (1..10);
+    }
+}
+";
+
+            const string fixedCode = @"using System;
+
+public class TestClass
+{
+    public void TestMethod()
+    {
+        var a = new int[0];
+        var b = a[..];
+        var range = 1..10;
+    }
+}
+";
+
+            DiagnosticResult[] expected =
+            {
+                Diagnostic(DiagnosticId).WithSpan(8, 19, 8, 23),
+                Diagnostic(ParenthesesDiagnosticId).WithLocation(8, 19),
+                Diagnostic(ParenthesesDiagnosticId).WithLocation(8, 22),
+                Diagnostic(DiagnosticId).WithSpan(9, 21, 9, 28),
+                Diagnostic(ParenthesesDiagnosticId).WithLocation(9, 21),
+                Diagnostic(ParenthesesDiagnosticId).WithLocation(9, 27),
+            };
+
+            var test = new CSharpTest(LanguageVersion.CSharp8)
+            {
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp31,
+                TestCode = testCode,
+                FixedCode = fixedCode,
+            };
+            test.ExpectedDiagnostics.AddRange(expected);
+
+            await test.RunAsync(CancellationToken.None).ConfigureAwait(false);
+        }
     }
 }

--- a/StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1119StatementMustNotUseUnnecessaryParenthesis.cs
+++ b/StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1119StatementMustNotUseUnnecessaryParenthesis.cs
@@ -117,6 +117,7 @@ namespace StyleCop.Analyzers.MaintainabilityRules
                     && !node.Expression.IsKind(SyntaxKind.CoalesceExpression)
                     && !node.Expression.IsKind(SyntaxKind.QueryExpression)
                     && !node.Expression.IsKind(SyntaxKind.AwaitExpression)
+                    && !node.Expression.IsKind(SyntaxKindEx.RangeExpression)
                     && !node.IsKind(SyntaxKind.ConstructorDeclaration))
                 {
                     if (node.Expression.IsKind(SyntaxKind.ConditionalAccessExpression)

